README.md

```diff
@@ -1,4 +1,208 @@
+  - [Basic concept](#basic-concept)
+  - [Side effects](#side-effects)
+  - [Implementation details](#implementation-details)
+  - [Logging](#logging)
+- [Sample app](#sample-app)
+
+## How to use
+
+### Gradle
+
+All release artifacts are hosted on [Maven Central](https://mvnrepository.com/repos/central):
+```gradle
+repositories {
+    mavenCentral()
+}
+
+implementation "com.freeletics.coredux:core:1.0.0"
+```
+
+If you want to use latest snapshot release from `master` branch:
+```gradle
+repositories {
+    // CoRedux snapshot repository
+    maven {
+        url "https://oss.sonatype.org/content/repositories/snapshots/"
+    }
+}
+
+implementation "com.freeletics.coredux:core:1.0.0-SNAPSHOT"
+```
+
+### Additional artifacts
+
+Following additional artifacts are also available:
+- `com.freeletics.coredux:log-android:1.0.0` - provides `LogSink` implementation, that utilizes `android.util.Log` class to print log messages
+
+## What is CoRedux
+
+CoRedux is a predictable state container, that is using same approach as [Redux](https://redux.js.org/) with
+ability to have additional side effects. 
+
+Implementation is based on [Kotlin](https://kotlinlang.org/) [coroutines](https://github.com/Kotlin/kotlinx.coroutines)
+and inspired by both [RxRedux](https://github.com/freeletics/rxredux) library 
+and [Coroutines in practice by Roman Elizarov](https://www.youtube.com/watch?v=a3agLJQ6vt8) KotlinConf2018 talk.
+
+### Basic concept
+
+Imagine, that we need to develop Calculator app that allows to do only following simple actions:
+```kotlin
+sealed class CalculatorAction {
+    data class Add(val value: Int) : CalculatorAction()
+    data class Deduct(val value: Int) : CalculatorAction()
+    data class Multiply(val value: Int) : CalculatorAction()
+    data class Divide(val value: Int) : CalculatorAction()
+}
+```
+
+With CoRedux you can create a store that will be a single source of current Calculator state 
+and will update it on each new incoming `CalculatorAction`:
+```kotlin
+val store = GlobalScope.createStore<Int, CalculatorAction>(
+    name = "Calculator",
+    initialState = 0,
+    reducer = { currentState, newAction ->
+        when (newAction) {
+            is CalculatorAction.Add -> currentState + newAction.value
+            is CalculatorAction.Deduct -> currentState - newAction.value
+            is CalculatorAction.Multiply -> currentState * newAction.value
+            is CalculatorAction.Divide -> currentState / newAction.value
+        }
+    }
+)
+```
+
+Where `reducer` is special function that takes current state, new action that wants to
+modify the state (_intention_) and returns result of state modification, that becomes new current state.
+
+To start receiving current state updates, Calculator UI implementation needs to subscribe to `store` instance:
+```kotlin
+store.subscribe { state -> updateUI(state) }
+```
+By default, when first subscription happens, `store` will emit `initialState` to it.
```

Comment:
It can be specifically configured by passing `launchMode`:
```kotlin
val store = scope.createStore(
    ...,
    launchMode = CoroutineStart.DEFAULT,
    ....
)
```

Default value is `CoroutineStart.LAZY`

Reply:
My point is, you can't disbale emitting the `initial state`, right?

> By default, when first subscription happens, `store` will emit `initialState` to it.

This means to me I somehow can configure the `store` not to submit the `initialState`

Reply:
Yes, `initialState` may not be dispatched if `launchMode = CoroutineStart.DEFAULT`, as there will be no `StateReceivers` at that moment, though `currentState` will be updated to `initialState`

Reply:
What about such phrasing:

With defaults (`launchMode = CoroutineStart.LAZY), on first subscription, `store` will emit `initialState` to it.